decorator.rb

```diff
@@ -0,0 +1,56 @@
+# Interface
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement this method'
+  end
+end
+
+# Base Decorator
+class NameDecorator < Nameable
```

Comment:
Great job @harshitajain06 🤝

- [ ] Kindly set your `attribute_accessor` for the base decorator just like in the screenshot below.
![attribute_accessor](https://user-images.githubusercontent.com/63917725/224246875-e5cac537-6304-4a1d-81fa-065307fb967b.PNG)